python/eups/Eups.py

```diff
@@ -1026,6 +1026,16 @@ def findProductFromVRO(self, name, version=None, versionExpr=None, eupsPathDirs=
                 if self.cmdName != "setup":
                     print >> sys.stderr, "%-15s %s" % (product.name, product.version)
 
+        # Hack around Product.getTable() slowness, by internally caching Product instances
+        # FIXME: Should be removed once the Product.getTable() bottleneck is resolved
+        try:
+            product = self._productCache[product]
```

Comment:
This presumably speeds things up by avoiding returning new product objects each time.  I would think it would be even better, and not much more complex, to memoize the complete `findProductFromVRO` function instead, unless it is expected that its outputs are not stable.